components/Palettes/src/UIColor+MDCPalettable.h

```diff
@@ -0,0 +1,28 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+#import "MDCPalettes.h"
+
+@protocol MDCPalettable <NSObject>
+
+@property(nonatomic, strong, nullable) UIColor *mdc_nextDarkerColorInPalette;
+@property(nonatomic, strong, nullable) UIColor *mdc_nextLighterColorInPalette;
+
+@end
+
+@interface UIColor (MDCPalettable) <MDCPalettable>
```

Comment:
This seems odd.
Can we just add the methods directly on the class?
What happens if self isn't in a palette?

Would this make more sense as methods in MDCPalette?

Reply:
Alternative version contained within [MDCPalette](https://github.com/material-components/material-components-ios/pull/2866/commits/b2d9c5937e64ca0adbf7f1a57b2e417ec56ff13a#diff-98bf3a9381b20b3567d58e353c79249eR218).